jardin/instrumentation/notifier.py

```diff
@@ -0,0 +1,88 @@
+import os
+import uuid
+import queue
+import atexit
+import threading
+
+from jardin import config as config
+from jardin.instrumentation.base_subscriber import BaseSubscriber
+
+class Notifer:
+  # We set maxsize to avoid leaking memory if the reporting thread is dead for any reason
+  # in report_event we use `put_nowait` to avoid blocking callers if the queue is full
+  _q = queue.Queue(maxsize=4096)
+  _subscribers = {}
+  _should_stop = False
+  _worker_count = 2
+  _workers = []
+
+  @classmethod
+  def is_enabled(self):
+      return os.environ.get("JARDIN_INSTRUMENTATION_ENABLED", "false") == "true"
+
+  @classmethod
+  def subscribe(self, handler):
+      if not self.is_enabled():
```

Comment:
Instead of sprinkling these `is_enabled` checks across the class, what about having an additional no-op implementation? Then use the env var at boot time to pick one implementation.

Reply:
Where would the "boot time" logic live?

Reply:
In [config.init()](https://github.com/instacart/jardin/blob/86339a14e14c87df72236fcf1a300ac8d95c54bc/jardin/config.py#L23-L44) or at the top-level of the module. I would prefer the former since the initialization order is better defined (although due to a recent change, it can be called in 2 different places 😦 ).

But now that you've made the change to synchronous callbacks (thank you 🙏), we shouldn't need the boolean enabled flag at all.

Reply:
I am still going to retain the enabled flag in case the subscriber we end up writing is terrible and we want to disable it quickly.